lib/ansible/modules/cloud/amazon/neptune_cluster.py

```diff
@@ -0,0 +1,356 @@
+            'skip_final_snapshot': dict(type='bool', default=False),
+            'final_snapshot_id': dict(type='str'),
+        },
+        supports_check_mode=False,
+    )
+
+    result = {
+        'changed': False,
+        'db_cluster_arn': ''
+    }
+
+    desired_state = module.params.get('state')
+
+    params = {}
+    if module.params.get('availability_zones'):
+        params['AvailabilityZones'] = module.params.get('availability_zones')
+    if module.params.get('retention_period'):
+        params['BackupRetentionPeriod'] = module.params.get('retention_period')
+    if module.params.get('character_set'):
+        params['CharacterSet'] = module.params.get('character_set')
+    if module.params.get('database_name'):
+        params['DatabaseName'] = module.params.get('database_name')
+    params['DBClusterIdentifier'] = module.params.get('name')
+    if module.params.get('parameter_group'):
+        params['DBClusterParameterGroupName'] = module.params.get('parameter_group')
+    if module.params.get('vpc_security_groups'):
+        params['VpcSecurityGroupIds'] = module.params.get('vpc_security_groups')
+    if module.params.get('db_subnet_group'):
+        params['DBSubnetGroupName'] = module.params.get('db_subnet_group')
+    params['Engine'] = module.params.get('db_engine')
+    if module.params.get('db_engine_version'):
+        params['EngineVersion'] = module.params.get('db_engine_version')
+    if module.params.get('port'):
+        params['Port'] = module.params.get('port')
+    if module.params.get('master_username'):
+        params['MasterUsername'] = module.params.get('master_username')
+    if module.params.get('master_password'):
+        params['MasterUserPassword'] = module.params.get('master_password')
+    if module.params.get('option_group'):
+        params['OptionGroupName'] = module.params.get('option_group')
+    if module.params.get('backup_window'):
+        params['PreferredBackupWindow'] = module.params.get('backup_window')
+    if module.params.get('maintenance_window'):
+        params['PreferredMaintenanceWindow'] = module.params.get('maintenance_window')
+    if module.params.get('replication_source'):
+        params['ReplicationSourceIdentifier'] = module.params.get('replication_source')
+    if module.params.get('encrypted'):
+        params['StorageEncrypted'] = module.params.get('encrypted')
+    if module.params.get('kms_key_id'):
+        params['KmsKeyId'] = module.params.get('kms_key_id')
+    if module.params.get('pre_signed_url'):
+        params['PreSignedUrl'] = module.params.get('pre_signed_url')
+    if module.params.get('enable_iam_auth'):
+        params['EnableIAMDatabaseAuthentication'] = module.params.get('enable_iam_auth')
+
+    client = module.client('neptune')
+
+    cluster_status = cluster_exists(client, module, params, result)
+
+    if desired_state == 'present':
+        if not cluster_status:
+            create_cluster(client, module, params, result)
```

Comment:
Rather than initialize `result` above and then passing it around and modifying the contents it would be more clear to return a result dict created inside create_cluster (and update_cluster and delete_cluster) `return {'db_cluster_arn': response['DBCluster']['DBClusterArn'], 'changed': True}` and `result = create_cluster(client, module, params)` (because right now the return statements do nothing), etc